File: src/main/java/uk/co/agware/carpet/change/tasks/ScriptTask.java
package uk.co.agware.carpet.change.tasks;

import uk.co.agware.carpet.database.DatabaseConnector;

/**
 * Created by Philip Ward <Philip.Ward@agware.com> on 14/06/2016.
 */
public class ScriptTask extends DatabaseTask {

    private String[] inputList;

    public ScriptTask(DatabaseConnector databaseConnector, String taskName, int taskOrder, String script, String delimiter) {
        super(databaseConnector, taskName, taskOrder);
        delimiter = delimiter == null || "".equals(delimiter) ? ";" : delimiter;
        this.inputList = script.split(delimiter);
    }


    @Override
    public boolean performTask() {
        for(String s : inputList){
            if(!databaseConnector.executeStatement(s.trim())){
                return false;
            }
        }
        return true;
    }
}

File: src/main/java/uk/co/agware/carpet/change/tasks/ScriptTask.kt
package uk.co.agware.carpet.change.tasks;

import com.fasterxml.jackson.annotation.JsonIgnore
import uk.co.agware.carpet.database.DatabaseConnector

/**
 * Created by Simon on 29/12/2016.
 */
class ScriptTask(@JsonIgnore override var databaseConnector: DatabaseConnector?, override var taskName: String, override var taskOrder: Int, val script: String, delimiter: String?): Task {

    val delimiter: String
    val inputList: List<String>

    init {
        this.delimiter = if (delimiter == null || "" == delimiter) ";" else delimiter
        this.inputList = this.script.split(this.delimiter.orEmpty())
    }

    @Override
    override fun performTask(): Boolean {
        this.inputList.forEach { s -> if(!this.databaseConnector!!.executeStatement(s.trim())) return false }
        return true
    }
}
